### PR TITLE
Rename new-api into backoffice API

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | This PR renames the prefix of the new APIs handled by ApiPlatform, but this `/api` prefix was already used by "legacy APIs" So we added a checker to make sure the path matches `^/api` AND is a controller related to ApiPlatform It's a temporary solution, the legacy APIs should be modified so that they use a different prefix
| Type?             | improvement
| Category?         |  WS 
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | CI and tests https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/5954875944 🟢 
| Fixed ticket?     | 
| Related PRs       | 
| Sponsor company   | PrestaShop SA

The below description is that wrote by @matks in https://github.com/PrestaShop/PrestaShop/pull/32484

## Description

> The new API built in `develop` has still no better name than “New API” and the endpoints are prefixed with `/new-api/`. The name “New API” is written in a few files in the software.
> 
> I think we all agree it’s not the best name ever 😆 
> 
> I suggest we rename the API as “Command API”  (as the API is supposed to plug on existing Commands from CQRS). URL prefix would become `/command-api`.
> 
> I considered the idea of using `/api/v2` for the URL prefix but PS webservices already use the `/api` prefix so it could create routing issues and confusion.
> 
> I also modify the version number from 1.0.0 to 0.1.0 to highlight the fact this is experimental and might change a lot in the next version.

### CODE REVIEW

**Comment by LouisAUTHIE** [comment]
Hi, do you want to wait for a chosen name to replace funkypapi ? Is it a draft ?

**Comment by nicosomb** [comment]
It was just a try to check if `somethingelseapi` works. We have to wait the decision of the tech council.

**Comment by LouisAUTHIE** [comment]
Ok great

File: src/PrestaShopBundle/Api/Api.php [hunk 1]
```diff
@@ -33,7 +33,7 @@
  */
 final class Api
 {
-    public const API_BASE_PATH = '/new-api';
+    public const API_BASE_PATH = '/funkypapi';
```
Comment (jolelievre) [inline src/PrestaShopBundle/Api/Api.php]: I'd simply use `/api` even if we already have existing APIs with this prefix I don't think it's a problem to use the same, the Swagger doc will only output the API platform endpoints anyway

Besides, even the token endpoint uses this prefix currently `/api/oauth/token` so why not uniformize this And the existing APIs will likely be replaced and removed by the new ones in time

**Review by matks — DISMISSED** [review]
@jolelievre decided on behalf of the tech council. It will be `/api`.

This PR must be updated to match the decision 🎉

**Comment by nicosomb** [comment]
Tests: https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5923066969

**Review by matthieu-rolland — DISMISSED** [review]

**Review by FabienPapet — DISMISSED** [review]

**Comment by matks** [comment]
Tests fail as there is 2 set of routing resources defined with `/api` prefix

⏩ some more work to do 😊

**Review by MhiriFaten — COMMENTED** [review]
Some checks were not successful,  could you check it please ?

**Comment by nicosomb** [comment]
👏 @jolelievre

**Comment by nicosomb** [comment]
Tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5960584371

**Review by mflasquin — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]
LGTM UI tests

**Review by M0rgan01 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by mflasquin** [comment]
It's OK for me :

image

image

**Comment by mflasquin** [comment]
thanks @nicosomb

**Pull request merged**